### ISSUE TITLE
VLC Connection Fails: "Cseq mismatch" & "only real/helix"

### ISSUE THREAD

**Issue by jlpoolen** [thirtythreeforty/neolink#12]
Using Window VLC Media Player for Windows 3.0.10 Vetinari on Windows 7.  Connection failed. 

Log reveal attempt to connect to the neolink proxy fails with "warning: Cseq mismatch, got 1, assumed 0" and "warning: only real/helix rtsp servers supported for now":

    main debug: creating access: rtsp://ares:8554/garage_west
    main debug:  (path: \\ares:8554\garage_west)
    main debug: looking for access module matching "rtsp": 26 candidates
    satip debug: try to open 'rtsp://ares:8554/garage_west'
    satip debug: connect to host 'ares'
    main debug: net: connecting to ares port 8554
    main debug: connection succeeded (socket = 1196)
    main debug: net: opening 0.0.0.0 datagram port 9478
    main debug: net: opening 0.0.0.0 datagram port 9479
    satip error: Failed to setup RTSP session
    main debug: net: connecting to ares port 8554
    main debug: connection succeeded (socket = 1228)
    access_realrtsp warning: Cseq mismatch, got 1, assumed 0
    access_realrtsp debug: rtsp connected
    access_realrtsp warning: only real/helix rtsp servers supported for now
    main debug: no access modules matched
    main debug: dead input

See https://pastebin.com/ALEh4z9d (30 days from 5/22/2020)

**Comment by thirtythreeforty** [comment]
Hey thanks for the report.

Which cameras are you using this with? Currently it's only tested with a handful of models - the B800/D800.

**Comment by thirtythreeforty** [comment]
That's not to say it won't work with the others listed in the readme (just added that section) - but it's untested :)

**Comment by jlpoolen** [comment]
Reolink RLC-420

**Comment by thirtythreeforty** [comment]
So for the RLC-420, they [support RTSP out of the box](https://support.reolink.com/hc/en-us/articles/360007010473-How-to-Live-View-Reolink-Cameras-via-VLC-Media-Player). For the time being, I am not planning on explicitly adding support for them. I would welcome contributions that add support, but since they're already usable with RTSP, it's fairly low priority.

**Comment by jlpoolen** [comment]
Have you noticed discrepancies in the protocol used by various models of the Reolink product line?  If so, then may I could capture some transmissions between the Reolink Windows client and the cameras using wireshark?

Another thing that came to mind, I saw in the log:

> [2020-05-21T17:04:12Z INFO  neolink::bc_protocol::connection] Ignoring uninteresting message ID 79
> [2020-05-21T17:04:12Z DEBUG neolink::bc_protocol::connection] Contents: Bc { meta: BcMeta { msg_id: 79, client_idx: 0, class: 0, encrypted: true }, body: ModernMsg(ModernMsg { xml: Some(BcXml { encryption: None, login_user: None, login_net: None, device_info: None, preview: None }), binary: None }) }

I wonder if those messages are alarm signals being sent to the monitoring client to start/stop recording?

**Comment by thirtythreeforty** [comment]
That's very possible. My camera sends msg 78 and 79 on connect, which Neolink ignores as you see.

Packet captures would be fantastic - if you can provide them it's possible I can add support. No guarantees, and like I said, your cameras also provide native RTSP.

**Comment by jlpoolen** [comment]
My ignorance, I did not realize that RLC-420 were supported out of the box.  I just installed Moonfire Nvr and am exploring it's ability to handle multiple feeds.  The port which seems to work is not 9000, but 554.  In the Moonfire configuration (ncurses interface) I populate the admin/password fields and then for the rtsp field I use (for a camera with and permanently assigned IP at 192.168.1.49):

     rtsp://192.168.1.49:554/h264Preview_01_main

I at least got something on the port 554 feed, but with the port 9000, I did not.  I recall playing around with these over a year ago and concluding that the port 9000 was not working and speculating that maybe they were not following the protocol.
